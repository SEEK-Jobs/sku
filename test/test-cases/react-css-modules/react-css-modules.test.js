const path = require('path');
const { promisify } = require('es6-promisify');
const rimrafAsync = promisify(require('rimraf'));
const dirContentsToObject = require('../../utils/dirContentsToObject');
const runSkuScriptInDir = require('../../utils/runSkuScriptInDir');
const appDir = path.resolve(__dirname, 'app');
const distDir = path.resolve(appDir, 'dist');

describe('react-css-modules', () => {
  beforeAll(async () => {
    await rimrafAsync(distDir);
    await runSkuScriptInDir('build', appDir);
  });

  it('should generate the expected files', async () => {
    const files = await dirContentsToObject(distDir);
    expect(files).toMatchSnapshot();
  });

  it('should handle Less and css.js in tests', async () => {
    const { childProcess } = await runSkuScriptInDir('test', appDir);
    expect(childProcess.exitCode).toEqual(0);
  });
});
