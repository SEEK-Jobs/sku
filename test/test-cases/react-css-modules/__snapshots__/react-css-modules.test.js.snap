// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react-css-modules should generate the expected files 1`] = `
Object {
  "index.html": "
  <!DOCTYPE html>
  <html>
    <head>
      <meta charset=\\"UTF-8\\">
      <title>My Awesome Project</title>
      <meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=1\\">
      <link rel=\\"stylesheet\\" type=\\"text/css\\" href=\\"/style.css\\" />
    </head>
    <body>
      <div id=\\"app\\"><div class=\\"_1brbpTo Jlnxw97\\" data-reactroot=\\"\\"><div class=\\"_3YtmnHj lwsthgY\\">Hello World</div></div></div>
      <script type=\\"text/javascript\\" src=\\"/main.js\\"></script>
    </body>
  </html>
",
  "main.js": "CONTENTS IGNORED IN SNAPSHOT TEST",
  "style.css": "._1brbpTo{display:-webkit-box;display:-webkit-flex;display:-ms-flexbox;display:flex}._1brbpTo ._3YtmnHj{font-size:32px}
.Jlnxw97{-webkit-box-align:center;-webkit-align-items:center;-ms-flex-align:center;align-items:center}.Jlnxw97 .lwsthgY{color:red}
",
}
`;
